### PR TITLE
Add country subdivisions to organizations

### LINKED ISSUE #175: Add country subdivision to organization
Add the following properties to the organizations endpoints:

- `country_subdivision` - `string ?`- ISO 3166-2 code of the organization's country subdivision.
- `country_subdivision_flag` - `array of FILE ?` - Flag of the country subdivision. A server is recommended to provide flags of size around 56x56 and 160x160. Only allowed mime types are image/*.  

This would be used e.g. at some NA contests that show state flags (i.e. country subdivisions), e.g.: https://nasouth24.kattis.com/contests/nasouth24d1/standings .

### PR DESCRIPTION
**Pull request by niemela**
Fixes #175 .

### CODE REVIEW

File: Contest_API.md [hunk 1]
```diff
@@ -1247,19 +1247,21 @@ organizations (i.e. return any organization\_ids).
...
+| Name                       | Type            | Description
+| :------------------------- | :-------------- | :----------
+| id                         | ID              | Identifier of the organization.
+| icpc\_id                   | string ?        | External identifier from ICPC CMS.
+| name                       | string          | Short display name of the organization.
+| formal\_name               | string ?        | Full organization name if too long for normal display purposes.
+| country                    | string ?        | [ISO 3166-1 alpha-3 code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3) of the organization's country.
+| country\_flag              | array of FILE ? | Flag of the country. A server is recommended to provide flags of size around 56x56 and 160x160. Only allowed mime types are image/\*.
+| country\_subdivision       | string ?        | [ISO 3166-2 code](https://en.wikipedia.org/wiki/ISO_3166-2) of the organization's country subdivision.
+| country\_subdivision\_flag | array of FILE   | Flag of the country subdivision. A server is recommended to provide flags of size around 56x56 and 160x160. Only allowed mime types are image/\*.
```
Comment (nickygerritsen) [inline Contest_API.md:1259]: Maybe add it to the example below?

**Review by nickygerritsen — DISMISSED** [review]

File: Contest_API.md (hunk 1 above)
Comment (johnbrvc) [inline Contest_API.md:1259]: Shouldn't the subdivision flag array be: `array of FILE ?`  that is, shouldn't it be optional?

File: Contest_API.md [hunk 2]
```diff
@@ -1247,19 +1247,21 @@ organizations (i.e. return any organization\_ids).
...
-| logo               | array of FILE ? | Logo of the organization. A server must provide logos of size 56x56 and 160x160 but may provide other sizes as well. Only allowed mime types are image/\*.
+| Name                       | Type            | Description
+| :------------------------- | :-------------- | :----------
+| id                         | ID              | Identifier of the organization.
+| icpc\_id                   | string ?        | External identifier from ICPC CMS.
+| name                       | string          | Short display name of the organization.
+| formal\_name               | string ?        | Full organization name if too long for normal display purposes.
+| country                    | string ?        | [ISO 3166-1 alpha-3 code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3) of the organization's country.
+| country\_flag              | array of FILE ? | Flag of the country. A server is recommended to provide flags of size around 56x56 and 160x160. Only allowed mime types are image/\*.
+| country\_subdivision       | string ?        | [ISO 3166-2 code](https://en.wikipedia.org/wiki/ISO_3166-2) of the organization's country subdivision.
```
Comment (eldering) [inline Contest_API.md:1258]: Maybe add for clarification:
```suggestion
| country\_subdivision       | string ?        | [ISO 3166-2 code](https://en.wikipedia.org/wiki/ISO_3166-2) of the organization's country subdivision (like state or province).
```

Comment (niemela) [reply Contest_API.md]: Added example matching the Wikipedia page.

**Review by johnbrvc — APPROVED** [review]

**Review by eldering — COMMENTED** [review]
Do we want a combination like
```
"country":"NLD","country_subdivision":"US-PA"
```
to be legal? If so, I think we should explicitly mention that it's possible, if not, we should add this constraint. Or say that you SHOULD NOT do this.

**Comment by niemela** [comment]
> Do we want a combination like
> 
> ```
> "country":"NLD","country_subdivision":"US-PA"
> ```

I'm a bit torn. On the one hand, that makes no sense (so, forbid it), on the other hand, why does it matter (so, allow it).

Maybe a SHOULD is the right balance, but then I don't love adding SHOULDs (which don't actually **do** anything), for things that are common sense anyway (and doesn't really hurt anything).

As I keep saying: not every dumb thing needs to be explicitly forbidden.

**Comment by deboer-tim** [comment]
I'm with @niemela on this one, I think it falls into the 'dumb / clearly wrong, but not directly harmful', so I wouldn't bother explicitly forbidding it.

That said, we could change the description to something like 'subdivision within the country' which makes it a bit more explicit?

**Review by eldering — APPROVED** [review]

**Pull request merged**